### ISSUE TITLE
"process out of memory" error 

### ISSUE THREAD

**Issue by jywarren** [RichardLitt/name-your-contributors#15]
Hi, really interested in using the lib, thanks! Got this error, though, after hanging for a few minutes:

```
(trusty)warren@localhost:~/sites/plots2$ name-your-contributors ipfs --since=2016-10-15T00:20:24Z
Got response
wrote issue_creators
wrote issue_commenters

<--- Last few GCs --->

  403290 ms: Mark-sweep 701.8 (738.5) -> 699.6 (738.5) MB, 8779.9 / 0 ms [allocation failure] [GC in old space requested].
  413027 ms: Mark-sweep 699.6 (738.5) -> 699.6 (738.5) MB, 9736.6 / 0 ms [allocation failure] [GC in old space requested].
  429222 ms: Mark-sweep 699.6 (738.5) -> 698.8 (738.5) MB, 16195.4 / 13 ms [last resort gc].
  441528 ms: Mark-sweep 698.8 (738.5) -> 697.9 (738.5) MB, 12302.8 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x29d65341 <JS Object>
    1: /* anonymous */(aka /* anonymous */) [/home/warren/.nvm/versions/node/v4.4.0/lib/node_modules/name-your-contributors/node_modules/octokat/dist/node/verb-methods.js:79] [pc=0x25676570] (this=0x29d080dd <undefined>,verbFunc=0x47f8f695 <JS Function module.exports.SimpleVerbs.verbs.create (SharedFunctionInfo 0x3a6b53e5)>,verbName=0x29d25431 <String[6]: create>)
    2: injectVerbMethods [/home/wa...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
Aborted (core dumped)
```

**Comment by jywarren** [comment]
That looks like a lot of data for just the past couple weeks of data; any ideas?

**Comment by RichardLitt** [comment]
Hey! Sorry for the massive delay getting back to you on this. Currently on vacation // hacking in Japan, haven't had much time. I think we've fixed this issue elsewhere, but I'm not sure I merged it in. Give me a couple of days to take a look at it.

**Comment by diasdavid** [comment]
Just encountered the same problem `FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory`

**Comment by RichardLitt** [comment]
Can you give me your command so I can test it?

**Comment by jywarren** [comment]
I'm using:

```js
var nameYourContributors = require('name-your-contributors');

nameYourContributors('ipfs', {
  since: '2016-12-01T00:00:00Z'
});
```

with prepended env variables in the `node nyc.js`. Thanks!

**Comment by jywarren** [comment]
But also the above-referenced direct CLI method:

```js
(trusty)warren@localhost:~/$ name-your-contributors ipfs --since=2016-10-15T00:20:24Z
```

**Comment by RichardLitt** [comment]
Hey! So! I found @tgetgood, and with his help, this issue is no longer an issue! 🎉 

This should now be fixed as of 3.0.0, which uses GraphQL to stop all overfetching issues. 

@jywarren Try it again? @diasdavid Feel free to use it for IPFS, too. :)

**Comment by jywarren** [comment]
This works great now. Thanks!
